rslabel_package/rslabel/ret.py

```diff
@@ -124,68 +165,79 @@ def app(args):
-
-                    log['labels_added'] += labels_added
-                    if labels_added > 0:
-                        log['images_labeled'] += 1
-        else:
-            for old, new in zip(previous_annotations, new_annotations):
-                if in_validation:
-                    log['images_validated'] += 1
-                    try:
-                        status = new['status']
+                        log['labels_added'] += labels_added
+                        if labels_added > 0:
+                            log['images_labeled'] += 1
+            else:
+                for old, new in zip(previous_annotations, new_annotations):
+                    if in_validation:
+                        log['images_validated'] += 1
                         try:
-                            old_status = old['status']
-                            if old_status != status:
+                            status = new['status']
+                            try:
+                                old_status = old['status']
+                                if old_status != status:
+                                    log['labels_validated'] += len(new['annotations'])
+                                    log['images_validated'] += 1
+                            except:
                                 log['labels_validated'] += len(new['annotations'])
                                 log['images_validated'] += 1
                         except:
-                            log['labels_validated'] += len(new['annotations'])
-                            log['images_validated'] += 1
-                    except:
-                        pass
-                else:
-                    labels_added = len(new['annotations']) - len(old['annotations'])
-                    if labels_added > 0:
-                        log['labels_added'] += labels_added
-                        log['images_labeled'] += 1
-
-        # Upload the JSON to the server.
-        with sftp.cd(dest_dir):
-            sftp.put(args.annotations)
+                            pass
+                    else:
+                        labels_added = len(new['annotations']) - len(old['annotations'])
+                        if labels_added > 0:
+                            log['labels_added'] += labels_added
+                            log['images_labeled'] += 1
+
+        # Upload the JSON to the server. Or tar if the images were bad
+        # in clarification proccess
+        if not delete:
+            with sftp.cd(dest_dir):
+                sftp.put(args.annotations)
+        else:
+            with sftp.cd(dest_dir):
+                sftp.put(tar_name)
+        os.remove(tar_name)
 
         # Move the tar from in_progress to the proper destination.
-        sftp.rename('{}/{}'.format(src_dir, tar_name),
-                    '{}/{}'.format(dest_dir, tar_name))
+        # or delete tar if images were bad in clarification
```

Comment:
We shouldn't need to delete or modify the TAR throughout the whole labeling process.